tests/conftest.py

```diff
@@ -339,23 +367,46 @@ def mesh_model(griduda):
 
 
 @pytest.fixture()
-def mock_driver(geodf: gpd.GeoDataFrame) -> GeoDataFrameDriver:
+def mock_resolver() -> MetaDataResolver:
+    class MockMetaDataResolver(MetaDataResolver):
+        def resolve(self, uri, *args, **kwargs):
+            return [uri]
+
+    resolver = MockMetaDataResolver()
+    return resolver
```

Comment:
You could try to use the `MockerFixture` of pytest and make a mock, instead of a full class.

```suggestion
def mock_resolver(mocker: MockerFixture) -> MetaDataResolver:
    return mocker.Mock(spec_set=MetaDataResolver, resolve=lambda _, uri, *_, **_: [uri])
```

Reply:
[SO](https://stackoverflow.com/questions/31709792/patching-a-class-yields-attributeerror-mock-object-has-no-attribute-when-acce) tells me we cannot autospec due to the dynamic nature of Pydantic classes.
We would have to create the properties ourselves. Perhaps it is worth it?